orm/encoding/ormkv/primary_key.go

```diff
@@ -0,0 +1,128 @@
+package ormkv
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+
+	"google.golang.org/protobuf/proto"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+// PrimaryKeyCodec is the codec for primary keys.
+type PrimaryKeyCodec struct {
+	*KeyCodec
+	msgType          protoreflect.MessageType
+	unmarshalOptions proto.UnmarshalOptions
+}
+
+// NewPrimaryKeyCodec creates a new PrimaryKeyCodec.
+func NewPrimaryKeyCodec(prefix []byte, msgType protoreflect.MessageType, fieldNames []protoreflect.Name, unmarshalOptions proto.UnmarshalOptions) (*PrimaryKeyCodec, error) {
```

Comment:
similar concern here, should we check prefix?

Reply:
nil is okay, i'll add doc comments